### ISSUE TITLE
Burning Board 4 only supports UTF-8 encoding

### ISSUE THREAD

**Issue by Sama34** [mybb/merge-system#134]
@dtdesign pointed out that Burning Board 4 only supports UTF-8 encoding (by default, at least) so things like `encode_to_utf8()` are unnecessary.

https://github.com/mybb/merge-system/commit/35021ffd81b67f5591205629b3b8afc088b1223f#diff-3d1e68484bf6f88f70629539863da0b9R55

**Comment by JN-Jones** [comment]
Note: this may affect other boards too. AFAIR IPB 4 asked only whether it should use utf8mb4 or utf8. Can't remember the other boards.

**Comment by dtdesign** [comment]
By the way, we strip all characters which do not fit into MySQL's `utf8`, for example you won't encounter any emojis as they require MySQL's `utf8mb4` to be storable.

**Comment by JN-Jones** [comment]
Yeah, we had that issue in the past (1.6.12 IIRC). Our db class handles that automatically now.

**Comment by JN-Jones** [comment]
This is related to #13. We need to discuss what we should do in those cases. I had already heated discussions with users where the database was corrupted (originally utf8 but exported as latin1) and they claimed that the original board seemed to work. Probably we should do something like I did with #143 and only show encodings which are really supported (we'd need to retest all installers/search files to test this).
Also we need to remember that we should support merges from utf8 to any of the encodings we support. However from what I've seen so far Dylan never considered those cases (like he never considered that the board isn't running on mysql) so probably it's easier to simply throw an error if the mybb board isn't using utf8.

@dtdesign: Gibt es eine Möglichkeit, das wir (als Team) eine Lizenz für eure Produkte bekommen? Bisher haben wir nur eine Lizenz für ein WBB 4, aber für keinerlei Erweiterungen oder 4.1. Details können wir gerne via Email oder PN ("Jones H" bei euch) klären.

**Comment by dtdesign** [comment]
@JN-Jones Feel free to approach us by email to woltlab@woltlab.com
